public/chat_flyout.tsx

```diff
@@ -22,7 +25,15 @@ interface ChatFlyoutProps {
 export const ChatFlyout = (props: ChatFlyoutProps) => {
   const chatContext = useChatContext();
   const chatHistoryPageLoadedRef = useRef(false);
-
+  const core = useCore();
+  // TODO: use DataSourceService to replace
+  const selectedDS = useObservable(
+    core.services.setupDeps?.dataSourceManagement?.dataSourceSelection?.getSelection$() ??
+      new BehaviorSubject(new Map())
```

Comment:
This new BehaviorSubject cause many re-render in my local env, shall we fix this?